spec/models/miq_request_workflow_spec.rb

```diff
@@ -152,4 +151,36 @@
       expect(workflow.provisioning_tab_list).to eq([{:name => "test3", :description => "test description 3"}])
     end
   end
+
+  context "'allowed_*' methods" do
+    let(:ems) { FactoryGirl.create(:ext_management_system) }
+
+    it "#allowed_clusters" do
+      cluster = FactoryGirl.create(:ems_cluster, :ems_id => ems.id)
+
+      allow(workflow).to receive(:get_source_and_targets).and_return(:ems => ems)
+      allow(Rbac).to receive(:search) do |arg|
+        expect(arg[:class]).to eq(EmsCluster)
+        expect(arg[:targets].size).to eq(1)
+        expect(arg[:results_format]).to eq(:objects)
+      end.and_return([[cluster], {}])
+      expect(workflow).to receive(:allowed_ci).with(:cluster, [:respool, :host, :folder], [cluster.id])
```

Comment:
Wouldn't it be easier to write a spec that just asserts the method definition matches an expected string? :worried: